src/Compat.jl

```diff
@@ -803,6 +803,224 @@ if VERSION < v"1.11.0-DEV.1579"
+                    "LogRange cannot start or stop at zero"))
+            elseif start < 0 || stop < 0
+                # log would throw, but _log_twice64_unchecked does not
+                throw(DomainError((start, stop),
+                    "LogRange does not accept negative numbers"))
+            elseif !isfinite(start) || !isfinite(stop)
+                throw(DomainError((start, stop),
+                    "LogRange is only defined for finite start & stop"))
+            elseif len < 0
+                throw(ArgumentError(string( # LazyString(
+                    "LogRange(", start, ", ", stop, ", ", len, "): can't have negative length")))
+            elseif len == 1 && start != stop
+                throw(ArgumentError(string( # LazyString(
+                    "LogRange(", start, ", ", stop, ", ", len, "): endpoints differ, while length is 1")))
+            end
+            ex = _logrange_extra(start, stop, len)
+            new{T,typeof(ex[1])}(start, stop, len, ex)
+        end
+    end
+
+    function LogRange{T}(start::Real, stop::Real, len::Integer) where {T}
+        LogRange{T}(convert(T, start), convert(T, stop), convert(Int, len))
+    end
+    function LogRange(start::Real, stop::Real, len::Integer)
+        T = float(promote_type(typeof(start), typeof(stop)))
+        LogRange{T}(convert(T, start), convert(T, stop), convert(Int, len))
+    end
+
+    Base.size(r::LogRange) = (r.len,)
+    Base.length(r::LogRange) = r.len
+
+    Base.first(r::LogRange) = r.start
+    Base.last(r::LogRange) = r.stop
+
+    function _logrange_extra(a::Real, b::Real, len::Int)
+        loga = log(1.0 * a)  # widen to at least Float64
+        logb = log(1.0 * b)
+        (loga/(len-1), logb/(len-1))
+    end
+
+    function Base.getindex(r::LogRange{T}, i::Int) where {T}
+        @inline
+        @boundscheck checkbounds(r, i)
+        i == 1 && return r.start
+        i == r.len && return r.stop
+        # Main path uses Math.exp_impl for TwicePrecision, but is not perfectly
+        # accurate, hence the special cases for endpoints above.
+        logx = (r.len-i) * r.extra[1] + (i-1) * r.extra[2]
+        x = _exp_allowing_twice64(logx)
+        return copysign(T(x), r.start)
+    end
+
+    function Base.show(io::IO, r::LogRange{T}) where {T}
+        print(io, "LogRange{", T, "}(")
+        ioc = IOContext(io, :typeinfo => T)
+        show(ioc, first(r))
+        print(io, ", ")
+        show(ioc, last(r))
+        print(io, ", ")
+        show(io, length(r))
+        print(io, ')')
+    end
+
+    # Display LogRange like LinRange -- PR widened signature of print_range to allow this
+    # function Base.show(io::IO, ::MIME"text/plain", r::LogRange)
+    #     isempty(r) && return show(io, r)
+    #     summary(io, r)
+    #     println(io, ":")
+    #     print_range(io, r, " ", ", ", "", " \u2026 ")
+    # end
+
+    _exp_allowing_twice64(x::Number) = exp(x)
+
+    if VERSION >= v"1.9"  # allows this high-precision path:
```

Comment:
For maximal pedantry
```suggestion
    if VERSION >= v"1.9.0-DEV.318"  # Julia PR #44717 allows this high-precision path:
```

Reply:
Sure, why not! Thanks for checking.